includes/class-sensei-learner-profiles.php

```diff
@@ -36,7 +36,32 @@ public function __construct() {
 
 		// Add class to body tag
 		add_filter( 'body_class', array( $this, 'learner_profile_body_class' ), 10, 1 );
-	} // End __construct()
+
+		// Scripts for frontend.
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+	} // End __construct().
+
+	/**
+	 * Enqueue frontend JavaScripts.
+	 *
+	 * @since  3.0.0
+	 * @access private
+	 */
+	public function enqueue_scripts() {
+		if ( ! Sensei_Utils::get_setting_as_flag( 'js_disable', 'sensei_settings_js_disable' ) ) {
+			global $wp;
+			global $post;
+			if ( preg_match( '/\/learner\//', add_query_arg( $wp->query_vars, home_url( $wp->request ) ) ) ||
+				'' !== get_query_var( 'learner_profile' ) ||
+				( is_a( $post, 'WP_Post' ) && has_shortcode( $post->post_content, 'usercourses' ) ) ) {
```

Comment:
This implies that the shortcode will exist on the learner profile page, which is highly unlikely. Shortcode-related logic should be kept in the shortcode.

Please see [here](https://github.com/Automattic/sensei/pull/2948/commits/523efcd33b789a85efe08c622e7acd52bfb58764) for a good example of how to enqueue these files on the learner profile page and on pages where the shortcode is used.

Reply:
Thanks for the code reference! Looks like #2948 does a cleaner job of addressing the problem. Should I go ahead and close this PR to prevent duplicate work?

Reply:
I think this one is a good intermediate solution. The other PR has issues that will need to be addressed.